packages/gatsby/src/utils/worker/child/load-config-and-plugins.ts

```diff
@@ -1 +1,17 @@
-export { loadConfigAndPlugins } from "../../../bootstrap/load-config-and-plugins"
+import { loadConfigAndPlugins as internalLoadConfigAndPlugins } from "../../../bootstrap/load-config-and-plugins"
+import { store } from "../../../redux"
+// this is just to drop return, as resulting config object can contain properties that can't be cloned
+// e.g. functions used in plugin options
+export async function loadConfigAndPlugins(
+  ...args: Parameters<typeof internalLoadConfigAndPlugins>
+): Promise<void> {
+  const [{ siteDirectory }] = args
+
+  store.dispatch({
```

Comment:
The worker needs some information on `store.getState().program`, for now we only added `directory`.